src/app/courses/courses-list/courses-list.component.ts

```diff
@@ -56,7 +38,26 @@ export class CoursesListComponent implements OnInit {
     console.log('clicked');
   }
 
-  onDeleteCourse(data: number) {
-    console.log(data);
+  onDeleteCourse(id: number) {
+    this.courseForDelete = this.coursesService.getCourseById(id);
+    if (this.courseForDelete) {
+      this.titleForDelete = this.courseForDelete.title;
+    }
+    this.showModal = true;
+  }
+
+  onCloseModal() {
+    console.log(this.courseForDelete);
+    if (this.courseForDelete) {
+      console.log(this.courseForDelete);
+      this.coursesService.removeCourse(this.courseForDelete.id);
+      this.courses = this.coursesService.getListCourses();
+    }
+    console.log('end');
+    this.showModal = false;
+  }
+
+  onCancelModal() {
```

Comment:
Please add the return type to the methods in this component

Reply:
done, will try to remember to add the return type consonantly